h/views/api/groups.py

```diff
@@ -81,6 +83,35 @@ def remove_member(group, request):
     return HTTPNoContent()
 
 
+@api_config(route_name='api.group_member',
+            request_method='POST',
+            link_name='group.member.add',
+            description='Add the current user to a group.',
+            effective_principals=security.Authenticated)
+def add_member(request):
+    """Add a member to a given group."""
+    client = request_auth_client(request)
+    validate_auth_client_authority(client, request.authority)
+
+    user_svc = request.find_service(name='user')
+    group_svc = request.find_service(name='group')
+    groupfinder_svc = request.find_service(IGroupService)
+
+    group = groupfinder_svc.find(request.matchdict['pubid'])
+    user = user_svc.fetch(request.matchdict['user'],
+                          client.authority)
+    if user is None:
+        raise HTTPNotFound()
+    if group is None:
+        raise HTTPNotFound()
+    if user.authority != group.authority:
+        raise HTTPNotFound()
```

Comment:
I don't think `HTTPNotFound` is the right exception to raise. Maybe `HttpException(404, "Group and user authorities don't match.")`

Reply:
Actually, it _is_ the correct error! If we raised another error, like a 403, that would be leaking information that a group does exist with that pubid but it belongs to another authority. That is a security issue. 404 is just fine and dandy here. From the perspective of the client's authority, the group _doesn't_ exist.